### ISSUE TITLE
Button triggering search modal missing aria-haspopup

### ISSUE THREAD

**Issue by swanrj08** [tomusborne/generatepress#663]
Add aria-haspopup="dialog" to the button so screen reader users are informed of the action when the button is activated. Additionally, include the aria-controls attribute to link the button to the ID of the modal being opened.
